fedimint-server/src/config.rs

```diff
@@ -35,32 +35,47 @@ use crate::{ReconnectPeerConnections, TlsTcpConnector};
 const DEFAULT_MAX_CLIENT_CONNECTIONS: u32 = 1000;
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
+/// All the serializable configuration for the fedimint server
 pub struct ServerConfig {
-    pub federation_name: String,
-    pub identity: PeerId,
-    pub hbbft_bind_addr: String,
-    pub api_bind_addr: String,
-    #[serde(with = "serde_tls_cert")]
-    pub tls_cert: rustls::Certificate,
+    /// Contains all configuration that needs to be the same for every server
+    pub consensus: ServerConfigConsensus,
+    /// Contains all configuration that is locally configurable and not secret
+    pub local: ServerConfigLocal,
+    /// Contains all configuration that is secret such as private key material
+    pub private: ServerConfigPrivate,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ServerConfigPrivate {
     #[serde(with = "serde_tls_key")]
     pub tls_key: rustls::PrivateKey,
-
-    pub peers: BTreeMap<PeerId, Peer>,
     #[serde(with = "serde_binary_human_readable")]
     pub hbbft_sks: SerdeSecret<hbbft::crypto::SecretKeyShare>,
     #[serde(with = "serde_binary_human_readable")]
-    pub hbbft_pk_set: hbbft::crypto::PublicKeySet,
+    pub epoch_sks: SerdeSecret<hbbft::crypto::SecretKeyShare>,
+    pub modules: BTreeMap<String, serde_json::Value>,
+}
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ServerConfigConsensus {
+    pub federation_name: String,
+    pub peers: BTreeMap<PeerId, Peer>,
```

Comment:
The `Peer` struct contains fields that I would not consider consensus critical, such as addresses under which peers are reachable (that might also be changed by the respective peer over the lifetime of the federation).

https://github.com/fedimint/fedimint/blob/4e700c1d26e465ed9694b05fffef4d831202330a/fedimint-server/src/config.rs#L67-L75

I think this requires further splitting (even if that's a bit inconvenient).

Reply:
True, should only the certs be consensus-critical?  I think it's important that peers agree on the set of certs to trust.

Reply:
Yes, seems like only the certs are relevant.